src/client/types.ts

```diff
@@ -0,0 +1,5 @@
+export type ServerResponse = {
+  data: any
```

Comment:
Suggestion without fully knowing if it's useful. What about using generics here?

```ts
export type ServerResponse<T = any> = {
  data: T
...
}
```

Avoids this: `ServerResponse & { data: RemoteItem }`

Reply:
Great catch, changing it.